Frontend/app/pages/enterpriseMap/enterpriseMap.component.ts

```diff
@@ -0,0 +1,108 @@
+import { Component, OnInit } from "@angular/core";
+import { SecretConfig } from "../../secretConfig";
+import { RouterExtensions, PageRoute } from "nativescript-angular/router";
+import "rxjs/add/operator/switchMap";
+import { LocalStorageService } from "../../shared/localStorageService";
+import { LocalDatabaseService } from "../../shared/localDatabaseService";
+import { Place } from "../../shared/place";
+ 
+// Displays a map associated with an enterprise to the user and allows
+// them to select a place.
+@Component({
+	selector: "map",
+	templateUrl: "pages/enterpriseMap/enterpriseMap.html",
+	styleUrls: ["pages/enterpriseMap/enterpriseMap-common.css", "pages/enterpriseMap/enterpriseMap.css"]
+})
+export class EnterpriseMapComponent implements OnInit
+{
+	mapboxAccessToken = SecretConfig.mapboxAccessToken;
+	enterpriseName: string;
+
+	private map: any
+
+	//Coordinates used for initial map view
+	private mapLatitude: Number;
+	private mapLongitude: Number;
+
+	constructor(private router : RouterExtensions,
+		private route: PageRoute,
+		private localDatabaseService: LocalDatabaseService)
+	{}
+	
+	//Stores Places and all their data
+	places: Array<Place> = [];
+	//Connects Place Id to the Participant Id
+	placesParticipant: Map<number, number> = new Map<number, number>();
+
+	enterpriseId: number;
+
+	ngOnInit()
+  	{
+		this.route.activatedRoute
+			.switchMap(activatedRoute => activatedRoute.params)
+			.forEach((params) => {
+			this.enterpriseId = +params['eId'];
+
+            this.localDatabaseService.getSavedEnterprise(this.enterpriseId).then(enterprise => {
+				this.enterpriseName = enterprise.name;
+				enterprise.participants.forEach((participant) => {
+
+					participant.places.forEach((place) => {
+						this.places.push(place);
+						this.placesParticipant[place.id] = participant.id;
+					})
+
+					var numOfParticipants = 0;
```

Comment:
This logic with the numOfParticipants variable and the setting of the map boundary looks a bit dodgy. It looks like each loop you reset 'numOfParticipants' to 0, which means the 'if' clause always gets executed. Hence, by this logic the map's boundaries will be that of the _last_ participant. You might be able to fix this by initialising the numOfParticipants variable outside of the loop.

Reply:
Fixed. I'm going to close this and make a new pull request with the work I've done today.